### PR TITLE
[MCC-652120] Incorporate protocol test suite as submodule

### PR DESCRIPTION
**Pull request by prajon84**
[Jira story](https://jira.mdsol.com/browse/MCC-652120)

This PR is for incorporating [mauth protocol test suite](https://github.com/mdsol/mauth-protocol-test-suite) into this "mauth-client-dotnet" repository as submodule.

It includes:

- Add `mauth-protocol-test-suite` as git submodule.
- Update contributing doc.
- Updated and verified all the unit tests run with `dotnet test`.

Please review:
@mdsol/architecture-enablement

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "mauth-protocol-test-suite"]
+	path = mauth-protocol-test-suite
+	url = git@github.com:mdsol/mauth-protocol-test-suite.git
```
Comment (ykitamura-mdsol) [inline .gitmodules:3]: Not sure for AppVeyor, but we need to change this URL for Travis

```suggestion
	url = https://github.com/mdsol/mauth-protocol-test-suite.git
```

https://github.com/mdsol/mauth-client-ruby/pull/45#pullrequestreview-553145146

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,7 +1,24 @@
 # Contributing
 
+## Cloning the Repo
+
+This repo contains the submodule `mauth-protocol-test-suite` so requires a flag when initially cloning in order to clone and init submodules.
+
+```
+git clone --recurse-submodules git@github.com:mdsol/mauth-client-ruby.git
```
Comment (ykitamura-mdsol) [inline CONTRIBUTING.md:8]:
```suggestion
git clone --recurse-submodules git@github.com:mdsol/mauth-client-dotnet.git
```

**Review by ykitamura-mdsol — COMMENTED** [review]
you might need to add
```yaml
install:
  git submodule update --init --recursive
```
to the `appveyor.yml` file.

https://www.appveyor.com/docs/how-to/private-git-sub-modules/

File: CONTRIBUTING.md (hunk 2 above)
Comment (prajon84) [reply CONTRIBUTING.md]: Oops!! Thanks for catching :)

File: .gitmodules (hunk 1 above)
Comment (prajon84) [reply .gitmodules]: Thanks. Yes Without "https" it said permission denied while cloning "mauth-protocol-test-suite".
Therefore, Updated to "https"

**Review by ykitamura-mdsol — APPROVED** [review]
looks good to me 👍

**Review by jfeltesse-mdsol — APPROVED** [review]

**Review by lschreck-mdsol — APPROVED** [review]

**Pull request merged**